ol/miner/src/block.rs

```diff
@@ -104,32 +105,87 @@ pub fn mine_and_submit(
         status_err!("Genesis block_0.json not found. Exiting.");
         std::process::exit(0);
     } else {
+        // the max block that has been succesfully submitted to client
+        let mut remote_height = current_block_number.unwrap();
+        // in the beginning, mining height is +1 of client block number
+        let mut mining_height = remote_height + 1; 
+
         // mine continuously from the last block in the file systems
-        let mut mining_height = current_block_number.unwrap() + 1;
         loop {
+            status_info!(
+                "Latest client block:",
+                format!("{}", remote_height)
+            );
+
             status_info!(format!("Block {}", mining_height), "Mining VDF Proof");
 
-            let block = mine_once(&config)?;
+            let block_result = mine_once(&config);
```

Comment:
If you just call backlog::process_backlog() below, you can simplify this match and just panic.

```
match mine_once(&config) {

                Ok(blk) => block = blk,
                Err(err) => {
                    status_warn!("Mining block {} failed: {}", mining_height, err);
                    continue
                }
            }
```

Reply:
Closed